### PR TITLE
counter: minor improvements in test and nRF RTC driver

### PR DESCRIPTION
**Pull request by nordic-krch**
**tests: drivers: counter: Add device name to zassert strings**
Prefix strings printed on zassert failure with device name.

**drivers: counter: Add helper macros for logging**
Added macros to simplify logging in the counter driver.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

### CODE REVIEW

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 1]
```diff
@@ -188,42 +198,45 @@ void test_single_shot_alarm_instance(const char *dev_name, bool set_top)
...
-		zassert_equal(0, err, "Counter failed to set top value\n");
+		zassert_equal(0, err,
+			     "%s: Counter failed to set top value\n", dev_name);
 
 		alarm_cfg.ticks = ticks + 1;
 		err = counter_set_channel_alarm(dev, 0, &alarm_cfg);
 		zassert_equal(-EINVAL, err,
-			      "Counter should return error because ticks"
-			      " exceeded the limit set alarm\n");
-		alarm_cfg.ticks = ticks - 1;
```
Comment (anangl) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c]: Was the removal of this line intentional? It is not mentioned in the commit message.

Comment (nordic-krch) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c]: no, i was cherry-picking it from other branch, probably automerge did that.

Comment (nordic-krch) [reply tests/drivers/counter/counter_basic_api/src/test_counter.c]: fixed

**Review by anangl — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 2]
```diff
@@ -93,10 +93,11 @@ static void counter_tear_down_instance(const char *dev_name)
 
 	err = counter_set_top_value(dev, counter_get_max_top_value(dev),
 				    NULL, NULL);
-	zassert_equal(0, err, "Setting top value to default failed\n");
+	zassert_equal(0, err,
+			"%s: Setting top value to default failed\n", dev_name);
```
Comment (nashif) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c]: you should remove \n

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 3]
```diff
@@ -93,10 +93,11 @@ static void counter_tear_down_instance(const char *dev_name)
...
 
 	err = counter_stop(dev);
-	zassert_equal(0, err, "Counter failed to stop\n");
+	zassert_equal(0, err, "%s: Counter failed to stop\n", dev_name);
```
Comment (nashif) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c]: ditto

File: tests/drivers/counter/counter_basic_api/src/test_counter.c [hunk 4]
```diff
@@ -106,12 +107,15 @@ static void test_all_instances(counter_test_func_t func)
...
+		/* Allow logs to be printed. */
+		k_sleep(100);
 	}
 }
 
 static void top_handler(struct device *dev, void *user_data)
 {
-	zassert_true(user_data == exp_user_data, "Unexpected callback\n");
+	zassert_true(user_data == exp_user_data,
+			"%s: Unexpected callback\n", dev->config->name);
```
Comment (nashif) [inline tests/drivers/counter/counter_basic_api/src/test_counter.c]: ditto and all below

**Review by nashif — DISMISSED** [review]

**Comment by nordic-krch** [comment]
@nashif removed all redundant `\n`.

**Comment by nordic-krch** [comment]
@nashif can you take a look and merge if possible?

**Comment by nordic-krch** [comment]
@nashif ping

**Comment by nordic-krch** [comment]
@nashif ping, is this change to trivial to get merged 😄 ?

**Review by carlescufi — APPROVED** [review]

**Pull request merged**